.werft/build.ts

```diff
@@ -289,6 +314,247 @@ interface DeploymentConfig {
+    setKubectlContextNamespace(namespace, metaEnv({ slice: installerSlices.SET_CONTEXT }));
+
+    // trigger certificate issuing
+    try {
+        werft.log(installerSlices.ISSUE_CERTIFICATES, "organizing a certificate for the preview environment...");
+
+        // trigger certificate issuing
+        await issueMetaCerts(namespace, domain);
+        await installMetaCertificates(namespace);
+        werft.done(installerSlices.ISSUE_CERTIFICATES);
+    } catch (err) {
+        werft.fail(installerSlices.ISSUE_CERTIFICATES, err);
+    }
+
+    // add the image pull secret to the namespcae if it doesn't exist
+    const hasPullSecret = (exec(`kubectl get secret ${IMAGE_PULL_SECRET_NAME} -n ${namespace}`, {slice: installerSlices.IMAGE_PULL_SECRET, dontCheckRc: true, silent: true })).code === 0;
+    if (!hasPullSecret) {
+        try {
+            werft.log(installerSlices.IMAGE_PULL_SECRET, "Adding the image pull secret to the namespace");
+            const auth = exec(`echo -n "_json_key:$(kubectl get secret ${IMAGE_PULL_SECRET_NAME} --namespace=keys -o yaml \
+                | yq r - data['.dockerconfigjson'] \
+                | base64 -d)" | base64 -w 0`, { silent: true }).stdout.trim();
+            const dockerConfig = { auths: { "eu.gcr.io": { auth: auth } } };
+            fs.writeFileSync(`./${IMAGE_PULL_SECRET_NAME}`, JSON.stringify(dockerConfig));
+            exec(`kubectl create secret docker-registry ${IMAGE_PULL_SECRET_NAME} -n ${namespace} --from-file=.dockerconfigjson=./${IMAGE_PULL_SECRET_NAME}`);
+            werft.done(installerSlices.IMAGE_PULL_SECRET);
+        }
+        catch (err) {
+            werft.fail(installerSlices.IMAGE_PULL_SECRET, err);
+        }
+    }
+
+    // download and init with the installer
+    try {
+        werft.log(installerSlices.INSTALLER_INIT, "Downloading installer and initializing config file");
+        exec(`docker run --entrypoint sh --rm eu.gcr.io/gitpod-core-dev/build/installer:${version} -c "cat /app/installer" > /tmp/installer`, {slice: installerSlices.INSTALLER_INIT});
+        exec(`chmod +x /tmp/installer`, {slice: installerSlices.INSTALLER_INIT});
+        exec(`/tmp/installer init > config.yaml`, {slice: installerSlices.INSTALLER_INIT});
+        werft.done(installerSlices.INSTALLER_INIT);
+    } catch (err) {
+        werft.fail(installerSlices.INSTALLER_INIT, err)
+    }
+
+    // prepare a proper config file
+    try {
+        werft.log(installerSlices.INSTALLER_RENDER, "Post process the base installer config file and render k8s manifests");
+        const PROJECT_NAME="gitpod-core-dev";
+        const CONTAINER_REGISTRY_URL=`eu.gcr.io/${PROJECT_NAME}/build/`;
+        const CONTAINERD_RUNTIME_DIR = "/var/lib/containerd/io.containerd.runtime.v2.task/k8s.io";
+
+        // get some values we need to customize the config and write them to file
+        exec(`yq r ./.werft/values.dev.yaml components.server.blockNewUsers \
```

Comment:
I understand that this is a concession to the interface problem around `values.dev.yaml` for as long as we have two ways to install on devstaging. I really hope we can get rid of this, soon. :slighty_smiling_face: